Added support for injecting MD files. Any extension starts with 'x-md…

TomerG711 opened the pull request:
…-*' will be treated as md file path, and loaded into 'md_*' property. The content of the MD file can be accessed in the templates.

This will allow us to create much more complex MD files, making Widdershins much more flexible.

MikeRalphson commented on lib/openapi3.js:
```diff
@@ -18,22 +18,35 @@ const dereference = require('reftools/lib/dereference.js').dereference;
+    for (let field in sourceSection) {
+        if (field.startsWith("x-md-")) {
+            //Replacing '-' so that field can be used in dot templates
+            let newFieldName = field.replace("x-md-", "md_");
```
Should we replace all `-` with `_` not just the first two? Either with a split/join or a `.replace`.

MikeRalphson commented on lib/openapi3.js:
```diff
@@ -18,22 +18,35 @@ const dereference = require('reftools/lib/dereference.js').dereference;
+        if (field.startsWith("x-md-")) {
+            //Replacing '-' so that field can be used in dot templates
+            let newFieldName = field.replace("x-md-", "md_");
+            destSection[newFieldName] = await fs.readFile(sourceSection[field], 'utf8');
```
Missing error handling on the `readFile`.

MikeRalphson commented on lib/openapi3.js:
```diff
@@ -104,8 +124,7 @@ function fakeProdCons(data) {
                     if (schema['x-widdershins-oldRef']) {
                         data.bodyParameter.refName = schema['x-widdershins-oldRef'].replace('#/components/schemas/', '');
-                    }
-                    else {
```
This is the first of many unrelated formatting changes even with GitHub diff set to ignore whitespace changes. Could we lose these please?

TomerG711 commented:
@MikeRalphson I've made the changes you asked. I added support to loading MD everywhere extensions are allowed, according to: https://swagger.io/docs/specification/openapi-extensions/.
Also made the field prefix configurable, and updated README.

TomerG711 commented:
@MikeRalphson Could you please take a look?

MikeRalphson commented:
@TomerG711 thanks for your patience. I'm evaluating a slightly different solution whereby we simply `recurse` through the input API definition (whether it is OpenAPI or AsyncAPI or anything else) and update the `x-md-` (or whatever prefix is specified) extensions - using code similar to what you have in the PR now.

Regardless of which approach we go with, I think we would need to support `fetch`ing of resources which are `http(s)` links or are relative to API definitions loaded from a URL.

MikeRalphson commented on lib/common.js:
```diff
@@ -526,6 +525,21 @@ function html(markdown,header,options) {
+                destSection[newFieldName] = md.render(fs.readFileSync(sourceSection[field],'utf8'));
+            }
+            catch (ex) {
+                throw ex;
```
Is there a reason to `catch` and then `throw` the same exception? If we're `catch`ing it, doing a `console.error(ex.message)` is probably sufficient.

TomerG711 commented:
So should I add support for fetching remote resources or should I leave it for you with the other solution?
